nodl_to_policy/_common/_profile.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional
+from lxml import etree
+try:
+    import importlib.resources as importlib_resources
+except ModuleNotFoundError:
+    import importlib_resources  # type: ignore
+
+
+__common_profile: etree.ElementTree = None
+__common_subscribe_topics: Optional[List] = None
+__common_publish_topics: Optional[List] = None
+__common_reply_services: Optional[List] = None
+__common_request_services: Optional[List] = None
+
+
+# TODO(aprotyas): Make use of `NodeType` here once the
+# presence/absence of lifecycle nodes are established
+# For now, only use `node.xml`, since NoDL does not describe Lifecycle nodes
+def common_profile() -> etree.ElementTree:
+    global __common_profile
+    if not __common_profile:
+        __common_profile = _get_profile('node.xml')
+    return __common_profile
+
+
+def common_subscribe_topics() -> List:
+    global __common_subscribe_topics
+    if not __common_subscribe_topics:
+        __common_subscribe_topics = _get_items_by_role('topics', 'subscribe')
+    return __common_subscribe_topics
+
+
+def common_publish_topics() -> List:
+    global __common_publish_topics
+    if not __common_publish_topics:
+        __common_publish_topics = _get_items_by_role('topics', 'publish')
+    return __common_publish_topics
+
+
+def common_reply_services() -> List:
+    global __common_reply_services
+    if not __common_reply_services:
+        __common_reply_services = _get_items_by_role('services', 'reply')
+    return __common_reply_services
+
+
+def common_request_services() -> List:
+    global __common_request_services
+    if not __common_request_services:
+        __common_request_services = _get_items_by_role('services', 'request')
+    return __common_request_services
+
+
+def _get_profile_path(filename):
+    with importlib_resources.path('nodl_to_policy._common', filename) as path:
+        return path
```

Comment:
This is a strange pattern; usually a `with` statement is used to keep a resource around for a while.  In this case, since we are returning it, the resource can be freed immediately on return.  What exactly is this code trying to accomplish?

Reply:
I was unsure about this too, but the `importlib` [API documentation](https://docs.python.org/3/library/importlib.html#importlib.resources.path) says "This function returns a context manager for use in a `with` statement", which is why I used this pattern.

Reply:
I see where you're coming from, though. I reckon this would just be clearer:
```suggestion
    return importlib_resources.path('nodl_to_policy._common', filename)
```

Reply:
The documentation says:

```
Exiting the context manager cleans up any temporary file created when the resource needs to be extracted from e.g. a zip file.
```

Which is why it is a context manager and not just a simple string.

Honestly, given how short this method is, I'll suggest just removing the method completely and open-coding the `with` statement in the one place you need it.  That way it will still be a context manager (and hence will cleanup after itself), and you'll still get the functionality you need.

Reply:
Fixed in 25d9db8.